plugins/Deflect/DeflectPlugin.cpp

```diff
@@ -184,6 +186,8 @@ class DeflectPlugin::Impl
         });
 
         _pollHandle->start(uvw::PollHandle::Event::READABLE);
+
+        _stream->setDisconnectedCallback([&] { _pollHandle->stop(); });
```

Comment:
capture `_pollHandle` by copy, so it's always safe to be valid

Reply:
Since `_pollHandle` is a member varaible, and [&] captures `this` be reference, this should be safe as well, shouldn't it?

Reply:
Not if `_pollHandle` is destructed before `_stream`? Also `&` is capturing more than you need.

Reply:
I tried [=, _pollHandle], but it doesn't compile. I'll try to get back to this after the weekend.

Reply:
just `[_pollHandle]` should work?!

Reply:
`[pollHandle=_pollHandle]` does work in fact

Reply:
while testing and looking at the code again, it probably should capture the pollhandle by reference and also `.reset()` it, so that `_closeStream()` does not stop the handle again. I at least faced some error during a reconnect phase:
```
[INFO ] Closing Deflect stream
braynsService: src/unix/core.c:896: uv__io_stop: Assertion `loop->watchers[w->fd] == w' failed.
Aborted
```